batteries/colorful.luau

```diff
@@ -1,27 +1,69 @@
+local process = (require)("@lute/process") -- temporary ()
+
+export type Styler = (text: string) -> string
+
+-- https://no-color.org/
+local function colorDisabled(): boolean
+	-- is it really okay to index env every style call?
+	return process.env.NO_COLOR ~= nil
```

Comment:
I'd suggest looking at how picocolors handles NO_COLOR. it also returns a generator function where you can explicitly specify if you want colors or not (but default returns the inferred preference) https://github.com/alexeyraspopov/picocolors/blob/main/picocolors.js

Reply:
Yeah, I think it'd be reasonable to return a function here, e.g.

```lua
function(colorsEnabled: boolean?)
    local colorsEnabled = colorsEnabled or (process.env.NO_COLOR ~= nil)

    return ...
end
```